src/components/NanoContract/NanoContractDetails.js

```diff
@@ -133,13 +137,53 @@ export const NanoContractDetails = ({ nc }) => {
             )}
             keyExtractor={(item) => item.txId}
             refreshing={isLoading}
-            onRefresh={handleMoreTransactions}
+            // Enables the pull gesture to get newer transactions
+            onRefresh={handleNewerTransactions}
+            // Enables a button to load more of older transactions until the end
+            // By reaching the end, the button ceases to render
+            ListFooterComponent={<LoadMoreButton lastTx={txHistory.slice(-1,).pop()} />}
+            extraData={[isLoading, error]}
           />
         )}
     </Wrapper>
   );
 };
 
+/**
+ * It show a button to 'Load More' transactions after the last one.
+ * It hides the button when the last transaction is the initialize.
+ *
+ * @param {Object} prop Properties object
+ * @param {{
+ *   ncId: string;
+ *   txId: string;
+ * }} prop.lastTx A transaction item from transaction history
+ */
+const LoadMoreButton = ({ lastTx }) => {
+  const dispatch = useDispatch();
+  const isInitializeTx = lastTx.ncMethod === 'initialize';
```

Comment:
You can continue using this to check if it already loaded the last tx, but now the API returns a new value in the response `has_more`

Reply:
Got it. Well, I prefer the current way. But if you want, I can open a KTLO to adapt the pagination to use the `has_more` property.